src/RageFileBasic.h

```diff
@@ -100,12 +100,20 @@ class RageFileObj: public RageFileBasic
 	virtual int GetFileSize() const = 0;
 	virtual int GetFD() { return -1; }
 	virtual RString GetDisplayPath() const { return RString(); }
-	virtual RageFileBasic *Copy() const { FAIL_M( "Copying unimplemented" ); }
+	virtual RageFileBasic* Copy() const
+	{
+		FAIL_M("Copying unimplemented");
+		return nullptr; // Return a default value - the return value is unused
```

Comment:
I should mention, I had to add the fake return values here so the game will compile.  Without them, compilation fails with only these two errors:

```
Severity	Code	Description	Project	File	Line	Suppression State	Details
Error	C4716	'RageFileObj::Copy': must return a value	ITGmania	RageFileBasic.h	106		
Error	C4716	'RageFileObj::SeekInternal': must return a value	ITGmania	RageFileBasic.h	112		

```

So really these should be fixed but I figured that is outside of the scope of the PR?

Reply:
Yeah this is fine. Weird that it didn't already have it but I think it should be returning something since it's a non-void function.

Reply:
> Yeah this is fine. Weird that it didn't already have it but I think it should be returning something since it's a non-void function.

Most likely the macro masked the issue by preventing the compiler from enforcing the return value requirement. I think that's why we now have a dozen or more warnings about 'not all control paths return a value' following this commit.